usr/lib/xtable/luaxtable.lua

```diff
@@ -0,0 +1,65 @@
+--
+-- SPDX-FileCopyrightText: (c) 2024 Mohammad Shehar Yaar Tausif <sheharyaar48@gmail.com>
+-- SPDX-License-Identifier: MIT OR GPL-2.0-only
+--
+
+local proto = {
+    ["UNSPEC"] = "NFPROTO_UNSPEC",
+    ["INET"] =  "NFPROTO_INET",
+	["IPV4"] =  "NFPROTO_IPV4",
+	["IPV6"] =  "NFPROTO_IPV6",
+	["ARP"] =  "NFPROTO_ARP",
+	["NETDEV"] =  "NFPROTO_NETDEV",
+	["BRIDGE"] =  "NFPROTO_BRIDGE",
+}
+
+local function write_file(filename, content)
+    local file = io.open(filename, "w")
+    if file == nil then
+        print("Failed to open file: "..filename)
+        return
+    end
+
+    file:write(content)
+    file:close()
+end
+
+local function read_file(filename)
+    local file = io.open(filename, "r")
+    if file == nil then
+        print("Failed to open file: "..filename)
+        return nil
+    end
+
+    local content = file:read("*a")
+    file:close()
+    return content
+end
+
+local function generate_module(ext_name, ext_rev, ext_family)
+    local luafile_tmpl = read_file("luaxtable.lua.tmpl") or ""
+    local makefile_tmpl = read_file("Makefile.tmpl") or ""
+    local cfile_tmpl = read_file("luaxtable.c.tmpl") or ""
+
+    local luafile = luafile_tmpl:gsub("<EXT_NAME>", ext_name)
+    local makefile = makefile_tmpl:gsub("<EXT_NAME>", ext_name)
+    local cfile = cfile_tmpl:gsub("<EXT_NAME>", ext_name)
+    cfile = cfile:gsub("<EXT_REV>", ext_rev or 1)
+    cfile = cfile:gsub("<EXT_FAMILY>", ext_family or "NFPROTO_UNSPEC")
+
+    write_file("libxt_"..ext_name..".c", cfile)
+    write_file("libxt_"..ext_name..".lua", luafile)
+    write_file("Makefile", makefile)
+end
+
+if #arg < 3 then
+    print("Usage: lua "..arg[0].." <module_name> <module_revision> <proto_family>")
```

Comment:
this all can be defined on the script using the Lua library.. you don't need to pass this in the command line..